packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -1,28 +1,38 @@
 import archiver from 'archiver';
 import fs from 'fs-extra';
 import glob from 'glob';
+import { prompt } from 'inquirer';
 import path from 'path';
 import _ from 'lodash';
 import { hashElement } from 'folder-hash';
+import { prevPermsQuestion } from './layerHelpers';
 import { getLayerMetadataFactory } from './layerParams';
 import { createLayerParametersFile, updateLayerCfnFile } from './storeResources';
 
 export async function packageLayer(context, resource) {
-  const resourcePath = path.join(context.amplify.pathManager.getBackendDirPath(), resource.category, resource.resourceName);
-  const layerData = getLayerMetadataFactory(context)(resource.resourceName);
+  const layerName = resource.resourceName;
+  const resourcePath = path.join(context.amplify.pathManager.getBackendDirPath(), resource.category, layerName);
+  const layerData = getLayerMetadataFactory(context)(layerName);
   let latestVersion: number = layerData.getLatestVersion();
   const curLayerHash = await hashLayerDir(resourcePath);
   const previousHash = layerData.getHash(latestVersion);
   const layerParameters = context.amplify.readJsonFile(path.join(resourcePath, 'layer-parameters.json'));
 
   if (previousHash && previousHash !== curLayerHash.hash) {
+    const prevPermissions = layerData.getVersion(latestVersion).permissions;
+    const defaultPermissions = [{ type: 'private' }];
+    let usePrevPermissions = true;
+    if (!_.isEqual(prevPermissions, defaultPermissions)) {
+      const { usePrevPerms } = await prompt(prevPermsQuestion(layerName));
```

Comment:
this is okay, but it would be nice to keep this function as just a packaging function and not ask any questions. Maybe this function could be broken up into `isNewVersion(layerData, curLayerHash): boolean` and `getNewVersionPermissions(prevPermissions: LayerPermission[]): LayerPermission[]`